ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualSignedTransaction.kt

```diff
@@ -0,0 +1,79 @@
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+import java.security.PublicKey
+
+/**
+ * Defines a signed Consensual transaction.
+ *
+ *  Comparing with [ConsensualLedgerTransaction]:
+ *  - It does not have access to the deserialized details.
+ *  - It has direct access to the signatures.
+ *  - It does not require a serializer.
+ *
+ * //TODO(below docs are from obsolete part, review them.)
+ * [ConsensualSignedTransaction] wraps the wire representation of the transaction. It contains one or more signatures,
+ * each one for a public key (including composite keys) that is mentioned inside a transaction state.
+ * [ConsensualSignedTransaction] is frequently passed around the network and stored. The identity of a transaction is
+ * the hash of Merkle root of the wrapped wire representation, therefore if you are storing data keyed by wire
+ * representations hash be aware that multiple different [ConsensualSignedTransaction]s may map to the same key (and
+ * they could be different in important ways, like validity!). The signatures on a ConsensualSignedTransaction
+ * might be invalid or missing: the type does not imply validity.
+ * A transaction ID should be the hash of the wrapped wire representation's Merkle tree root.
+ * Thus adding or removing a signature does not change it.
+ */
+@DoNotImplement
+@CordaSerializable
+interface ConsensualSignedTransaction {
+    /**
+     * @property id The ID of the transaction.
+     */
+    val id: SecureHash
+
+    /**
+     * @property signatures The signatures that have been applied to the transaction.
+     */
+    val signatures: List<DigitalSignatureAndMetadata>
+
+    /**
+     * Converts the current [ConsensualSignedTransaction] into a [ConsensualLedgerTransaction].
+     *
+     * @param serializer The [SerializationService] required to convert the current transaction.
+     * @return Returns a [ConsensualLedgerTransaction] from the current signed transaction.
+     */
+    fun toLedgerTransaction(serializer: SerializationService): ConsensualLedgerTransaction
+
+    /**
+     * Adds the specified signature to the current [ConsensualSignedTransaction].
+     *
+     * @param signature The signature and metadata to add to the current [ConsensualSignedTransaction].
+     * @return Returns a new [ConsensualSignedTransaction] containing the applied signature.
+     */
+    fun addSignature(signature: DigitalSignatureAndMetadata): ConsensualSignedTransaction
+
+    /**
+     * Adds the specified signatures to the current [ConsensualSignedTransaction].
+     *
+     * @param signatures The signatures and metadata to add to the current [ConsensualSignedTransaction].
+     * @return Returns a new [ConsensualSignedTransaction] containing the applied signatures.
+     */
+    fun addSignatures(signatures: Iterable<DigitalSignatureAndMetadata>): ConsensualSignedTransaction
```

Comment:
These APIs feel a bit strange to me. In the builder we ask for a public key and then generate a signed transaction (presumably with that signature attached), but then if you want to add more you do it by generating a signature object and putting it on the tx. Can you add a signature over some other data for example? Is that useful?

Reply:
These functions came from here:
https://github.com/corda/corda5/blob/40b67c5cff50c69a1e1b792ad11b8d7d2acfbccc/ledger/ledger/src/main/kotlin/net/corda/v5/ledger/transactions/SignedTransaction.kt#L45

But I think you are right and gathering the signatures in different ways does not sound consistent.

I am not too sure about your questions.

Reply:
@JamesHR3 suggested on our call to replace addSignature with something like this:
```
fun sign(publickey: ...)
```
to be more consistent with `signInitial()`